### PR TITLE
Add topic for diagnostic IDs

### PR DESCRIPTION
**Pull request by terrajobst**
- [x] Add links from [C# coding guidelines](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions)

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/fundamentals/coding-style/coding-conventions.md](https://github.com/dotnet/docs/blob/7255a930acf0d79fdb946f8ae46fc0a8e4457cba/docs/csharp/fundamentals/coding-style/coding-conventions.md) | [Common C# code conventions](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions?branch=pr-en-us-37877) |
| [docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md](https://github.com/dotnet/docs/blob/7255a930acf0d79fdb946f8ae46fc0a8e4457cba/docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md) | [Choosing diagnostic IDs](https://review.learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/choosing-diagnostic-ids?branch=pr-en-us-37877) |

### CODE REVIEW

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (terrajobst) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:39]: Is this the right way to link to APIs?

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (terrajobst) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:5]: No sure what those are used for; I also added `analyzers`

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (333fred) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:37]: This paragraph feels a bit too informal for a docs page.

```suggestion
Don't limit your prefix to two-characters (such as `CSXXX`and `CAXXXX`).
Instead, use a longer prefix to avoid conflicts. For example, the `System.*`
diagnostics use `SYSLIB` as their prefix.
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (333fred) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:33]:
```suggestion
> It's a source breaking change to change diagnostic IDs, as existing
> suppressions would be ignored.
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:8]:
```suggestion
# Choose diagnostic IDs
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:13]:
```suggestion
The IDs are surfaced from various APIs, such as:
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:15]:
```suggestion
* <xref:Microsoft.CodeAnalysis.DiagnosticDescriptor.Id?displayProperty=nameWithType>
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:16]:
```suggestion
* <xref:System.ObsoleteAttribute.DiagnosticId?displayProperty=nameWithType>
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:17]:
```suggestion
* <xref:System.Diagnostics.CodeAnalysis.ExperimentalAttribute.DiagnosticId?displayProperty=nameWithType>
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:19]:
```suggestion
Diagnostic IDs are also used as identifiers in source, for example, from [#pragma warning disable]
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:27]:
```suggestion
* Diagnostic IDs should be of the form `<PREFIX><number>`
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:29]:
```suggestion
  - The number represents the specific diagnostic
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:28]:
```suggestion
  - The prefix is specific to your project
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:32]:
```suggestion
> It's a source breaking change to change diagnostic IDs. For example, existing
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:33]:
```suggestion
> suppressions would be ignored if the ID changed.
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:35]:
```suggestion
To avoid ID conflicts, you generally don't want to limit your prefix to two-characters (such as `CSXXX`
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:36]:
```suggestion
or `CAXXXX`). Rather, use a longer prefix. For example, the `System.*`
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:37]:
```suggestion
diagnostics use the prefix `SYSLIB`.
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [reply docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:39]: No, I put the correct xrefs earlier in the review.

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:42]:
```suggestion
[#pragma warning disable]: ../language-reference/preprocessor-directives.md#pragma-warning
```

File: docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md
Comment (gewarren) [inline docs/csharp/roslyn-sdk/choosing-diagnostic-ids.md:43]:
```suggestion
[.editorconfig]: ../../fundamentals/code-analysis/code-style-rule-options.md
```

**Review by gewarren — APPROVED** [review]

**Comment by terrajobst** [comment]
> [ ] Add links from APIs

Looks like those would be in a different repo, so I guess we'll have to wait until this PR is merged.

**Pull request merged**
